### PR TITLE
Feat: Added History Page

### PR DESCRIPTION
**Pull request by sainath7878**
Added History Page

Files to be reviewed:

- [src/components/historyCard/HistoryCard.jsx](https://github.com/sainath7878/joddtube/compare/HistoryPage?expand=1#diff-fdb1f318f8ceafc6d4bab8e8476c4db6e58a5283856af1e7ce1bf54f0d603abe)
- [src/context/videoContext.js](https://github.com/sainath7878/joddtube/compare/HistoryPage?expand=1#diff-a5022f1409674001e58ff7ba0919d901a3ddc026f0c9a9fa0b4be5fab2233587)
- [src/pages/historyPage/HistoryPage.jsx](https://github.com/sainath7878/joddtube/compare/HistoryPage?expand=1#diff-c37039b19a499bf87d80cb7f84cd53be61d7007752ccf1f40565079e189248e9)
- [src/pages/videoPage/VideoPage.jsx](https://github.com/sainath7878/joddtube/compare/HistoryPage?expand=1#diff-c3da235c1d2cbf2c1c6bb4d697cc0f3353d4c1eb22992aedb21eafcd9916267e)
- [src/reducer/videoReducer.js](https://github.com/sainath7878/joddtube/compare/HistoryPage?expand=1#diff-2153ec3493046853fe510118e935cd85510bae579392cb294f33ee9b73ec1d23)

### Thanks for reviewing!

### CODE REVIEW

**Review by anik31 — APPROVED** [review]
Everything looks good. The code is very clean and readable. History functionality is working fine. Good that you've fixed the watch-later issue.

File: src/components/historyCard/HistoryCard.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    imgSrc,
+    name,
+    thumbnail,
+    title,
+    creator,
+    views,
+    uploaded,
+    length,
+    verified,
+  } = video;
```
Comment (Maianki) [inline src/components/historyCard/HistoryCard.jsx:19]: appreciation: good use of destructuring.

File: src/components/historyCard/HistoryCard.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  const { removeFromHistoryHandler } = useVideos();
+
+  return (
+    <div className={styles.cardContainer}>
+      <Link to={`/video/${_id}`} className={`card ${styles.card} `}>
+        <img
+          src={imgSrc}
+          alt={title}
+          loading="lazy"
```
Comment (Maianki) [inline src/components/historyCard/HistoryCard.jsx:29]: appreciation: use of lazy loading in images.

File: src/context/videoContext.js [hunk 3]
```diff
@@ -119,9 +110,55 @@ function VideoProvider({ children }) {
...
+                        authorization: encodedToken
+                    }
+                }
+            )
+            if (response.status === 200) {
+                videoDispatch({
+                    type: "SET_HISTORY_VIDEOS",
+                    payload: { historyVideos: response.data.history },
+                });
+                toast.error("History Cleared")
```
Comment (Maianki) [inline src/context/videoContext.js:150]: appreciation: use of toast for displaying messages/alerts

**Review by Maianki — APPROVED** [review]
Good job. The code is clean and very well structured.

**Pull request merged**